python/ray/air/util/tensor_extensions/utils.py

```diff
@@ -28,16 +30,52 @@ def _is_ndarray_variable_shaped_tensor(arr: np.ndarray) -> bool:
     return True
 
 
-def _create_possibly_ragged_ndarray(
+@PublicAPI(stability="alpha")
+def create_possibly_ragged_ndarray(
     values: Union[np.ndarray, "ABCSeries", Sequence[Any]]
 ) -> np.ndarray:
-    """
-    Create a possibly ragged ndarray.
-    Using the np.array() constructor will fail to construct a ragged ndarray that has a
-    uniform first dimension (e.g. uniform channel dimension in imagery). This function
-    catches this failure and tries a create-and-fill method to construct the ragged
-    ndarray.
-    """
+    """Create a possibly ragged ndarray.
```

Comment:
Nit: Our use of "ragged" here is a bit lenient; we only support collections of variable-sized ndarrays, where all subndarrays are uniform (non-ragged). But this is probably fine for now.